include/swift/SIL/OwnershipUtils.h

```diff
@@ -569,6 +613,11 @@ struct InteriorPointerOperand {
     return true;
   }
 
+  /// Return the base BorrowedValue of the incoming value's operand.
+  Optional<BorrowedValue> getSingleBaseValue() const {
```

Comment:
It's very efficient, and works very well in practice for `SILValue()` to have an invalid default state. Why introduce a different programming model for `BorrowedValue` involving Optional when you want them to be used the same way?

Reply:
I think what you are saying here makes sense. I'll change that.